rclcpp/test/rclcpp/test_node.cpp

```diff
@@ -2556,6 +2556,7 @@ TEST_F(TestNode, get_publishers_subscriptions_info_by_topic) {
   };
   rclcpp::QoS qos = rclcpp::QoS(qos_initialization, rmw_qos_profile_default);
   auto publisher = node->create_publisher<test_msgs::msg::BasicTypes>(topic_name, qos);
+  std::this_thread::sleep_for(std::chrono::milliseconds(1));
```

Comment:
this should be all synchronous, why is necessary to wait?

Reply:
I was getting random failures in this test. I didn't have it anymore when adding this sleep.

Reply:
We've seen this elsewhere. The problem is that depending on the backend DDS implementation, creating a publisher and then immediately trying to get info on it may or may not work. That's because the backend may create the topic asynchronously.

You'll probably want a solution similar to #1606.

Reply:
Agreed, I'd say remove this as it is (AFAIK) unrelated to this change, and also because it isn't a scalable solution.